GlotPress/GlotPress-WP
# Improve Site Tour UI

## Description

Improve the current UI:
- [x] Set a uniform color system, based on the `gp-primary-color`
- [x] Add i18n
- [x] Make the popover border radius concentric with the header radius
- [x] Use negative margins to compensate the spacing added by the popover pulse wrapper to avoid UI shifiting
- [x] Fix the UI shift from the pulse icon
- [x] Enqueue gp-tour and popover on Projects templates


## Testing Instructions

Add the below filter to test the Tours for Projects, Translation filters and editing, as below screenshots:
```
add_filter(
	'gp_tour',
	function( $tour ) {

		$new_tour = json_decode(
			'{
				"ui-intro" : [
					{
						"title": "GlotPress Introduction Tour",
						"color" : "#826eb4"
					},
					{
						"selector": ".gp-content p.actionlist a[href=\"/glotpress/projects/-new/\"]",
						"html": "Click here to create a new translation Project."
					},
					{
						"selector": ".gp-content p.actionlist a[href=\"/glotpress/languages/\"]",
						"html": "Click here to check your projects per language."
					},
					{
						"selector": "#project-actions-toggle",
						"html": "Click here to open your project actions."
					},
					{
						"reveal": "#project-actions-toggle",
						"selector": ".gp-content p.actionlist a[href=\"/glotpress/projects/test/import-originals/\"]",
						"html": "Click here to import originals"
					}
				],
				"filters-guide" : [
					{
						"title": "UI Introduction Tour",
						"color" : "#3939c7"
					},
					{
						"selector": ".revealing.filter",
						"html": "Click here to reveal the search field"
					},
					{
						"reveal": ".revealing.filter",
						"selector": ".filters-expanded input.is-primary",
						"html": "Click here to search"
					}
				],
				"translation-guide" : [
					{
						"title": "How to translate",
						"color" : "#f939c7"
					},
					{
						"selector": ".strings .original",
						"html": "This is the English text"
					},
					{
						"selector": ".strings .textareas",
						"html": "Here you can enter your translation"
					},
					{
						"selector": ".actions .is-primary",
						"html": "Submit your translation"
					}
				]
			}',
			true
		);

		return array_merge( $tour, $new_tour );
	}
);
```

## Screenshots

As published in the main PR [suggestion](https://github.com/GlotPress/GlotPress/pull/1632#issuecomment-1669664624):

![imagem](https://github.com/GlotPress/GlotPress/assets/7371591/a2f4aa8e-3b1e-4876-b1f9-cc72318e0b56)
![imagem](https://github.com/GlotPress/GlotPress/assets/7371591/0648ddd9-56e7-4cc5-b254-590334706c7b)
![imagem](https://github.com/GlotPress/GlotPress/assets/7371591/d98deb9f-b5d2-4b30-8235-9ef47666343f)
![imagem](https://github.com/GlotPress/GlotPress/assets/7371591/877f605c-85db-415e-87c6-982a207d2d16)
